### DISCUSSION TITLE
Why does the image extension not resize animated GIFs?

### DISCUSSION THREAD

**Discussion by glow12121** [datenstrom/community#771] [Ask a question] [answered]
Hi,

I have been experimenting with using animated GIFs in Yellow and it seems they won't play when upscaled. For example, this code makes an animated GIF a static image:

`[image picture.gif comment center 150%]`

**Comment by markseuffert** [comment, marked as answer]
The [image extension](https://github.com/annaesvensson/yellow-image) uses the GD image library and what you describe appears to be a [known problem](https://github.com/libgd/libgd/issues/648#issuecomment-773057651). It's a general lack of handling image frames. As soon as animation is supported in the GD image library, it will become available in Datenstrom Yellow too. As a workaround, resize animated GIFs in your favourite image editor.

**Reply by glow12121** [reply]
Thank you. I have also found this workaround, just using this code:

```
<img src="/media/images/picture.gif" alt="picture.gif"  width="75%" />
```

The percentage is that of the parent structure.

**Reply by markseuffert** [reply]
If you're interested in some background the image extension solves some problems that Markdown and HTML can't solve. For example the HTML [img element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img) expects width and height in pixel and they are [not intended to be used to stretch the image](https://www.w3.org/TR/2021/SPSD-html52-20210128/semantics-embedded-content.html#sec-dimension-attributes). I would suggest CSS when it comes to image transformation, but see yourself:

https://stackoverflow.com/questions/49614922/how-do-i-double-the-size-of-an-image-on-a-web-page
https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/scale

One doesn't have to suffer that much. [Here's an online image editor](https://ezgif.com), hope it helps.

**Reply by glow12121** [reply]
Using this online editor, I discovered that the most lightweight format for animated web images is webp. It's amazing how small it is. And a Linux video editor that can export to it is called Shotcut. This worked for me.

**Reply by markseuffert** [reply]
I heard you like small files? You should try converting GIF to MP4 video.

https://user-images.githubusercontent.com/2429165/160802010-05f6808c-670f-4415-80e4-131114aa164c.mp4

**Reply by glow12121** [reply]
>  mandalorian.mp4

:) Can Yellow handle mp4's so they autoplay and loop?

**Reply by markseuffert** [reply]
Give it a try, use your HTML powers. As an alternative make a new extension with the [API for developers](https://datenstrom.se/yellow/help/api-for-developers). As a developer you can extend Markdown with shortcuts, which can generate exactly the HTML code you want. Maybe with an option to autoplay and loop video? Maybe with an option to convert uploaded GIFs to MP4s? The API has you covered. 😁

**Reply by markseuffert** [reply]
As a starting point for new developers you can have a look at the HTML audio extension and convert it for HTML video:

https://github.com/schulle4u/yellow-audio

For more information see the API, hope this helps.

**Reply by glow12121** [reply]
mp4 is even better than webp. The problem I faced is the video won't autoplay. I tried this code in the browsers Vivaldi and Falkon. There's a still image the size of the video; I can make it play if I right-click, choose "Controls", and click "Play" button.

```
<video autoplay muted loop>
  <source src="/media/images/video.mp4" type="video/mp4" />
  Your browser does not support the video tag.
</video>
```
